app/controllers/candidate_interface/gcse/science/grade_controller.rb

```diff
@@ -1,39 +1,108 @@
 module CandidateInterface
   class Gcse::Science::GradeController < CandidateInterfaceController
     include Gcse::GradeControllerConcern
-
     before_action :redirect_to_dashboard_if_submitted
     before_action :set_subject
 
     def edit
-      @application_qualification = details_form
-      @qualification_type = details_form.qualification.qualification_type
+      @gcse_grade_form = science_gcse_grade_form
+      @qualification_type = gcse_science_qualification.qualification_type
+
+      render view_path
     end
 
     def update
-      @qualification_type = details_form.qualification.qualification_type
-
-      details_form.grade = details_params[:grade]
-      details_form.other_grade = details_params[:other_grade]
-
-      @application_qualification = details_form.save_grade
+      @gcse_grade_form = science_gcse_grade_form.tap do |form|
+        form.subject = science_details_params[:gcse_science] || @subject
+        form.grade = assign_grade
+        form.biology_grade = science_details_params[:biology_grade]
+        form.physics_grade = science_details_params[:physics_grade]
+        form.chemistry_grade = science_details_params[:chemistry_grade]
+        form.gcse_science = science_details_params[:gcse_science]
+      end
```

Comment:
Is there a reason we're not passing the params to the model and having it deal with munge all the data? I feel like we could pull quite a bit of logic out of the controller.

Reply:
Yeah was thinking about that also ...
Possibly the way to go would be to have something like `ScienceGcseGradeForm.assign_attributes_from(params)`
I'll have a play ...